src/test/test-utils.ts

```diff
@@ -101,3 +103,22 @@ function isReadonlyArray(maybeArr: any): maybeArr is ReadonlyArray<any> {
 function isWarning(wOrS: Warning|SourceRange): wOrS is Warning {
   return 'code' in wOrS;
 }
+
+export async function runScanner(
+    analyzer: Analyzer,
+    scanner: Scanner<ParsedDocument, any, any>,
+    url: string): Promise<{features: ScannedFeature[], warnings: Warning[]}> {
+  const context = await analyzer['_analysisComplete'];
+  const resolvedUrl = analyzer.resolveUrl(url);
+  const parsedDocument = await context['_parse'](resolvedUrl);
+  return scan(parsedDocument, [scanner]);
+}
+
+export async function runScannerOnContents(
```

Comment:
These names are very similar. Document or disambiguate a bit? I could even see a way to combine them with optional arguments.

Reply:
+1, documented.